### ISSUE TITLE
[Feature idea] Sending "Server files" on the agents

### ISSUE THREAD

**Issue by cristianadrielbraun** [Ylianst/MeshCentral#2482]
Hi, everyone.

I really want to find a reason to use the "Server Files" (or "My files") section, but I couldn't find one.

What I would like to do is simple:
It would be great if we can use the stored files as a repository to send it to the clients. I mean, at the same way be click on upload and select a local file, just another button to do the same, but choosing from the Server Files.
We even already have them categorized by groups and all!

There is the [Files Distribution](https://github.com/ryanblenis/MeshCentral-FileDistribution) plugin (very cool actually), 

But is not exactly the same, since this plugin will automatically send a file to a fixed folder, on a fixed device, according to a schedule.

But I feel this feature would be really easy to add. Given the categories and the existing plugin, I think the feature is almost there, we are just missing the button to do it

**Comment by folosleg** [comment]
Let me revive this thread...
I'm puzzled about the usecase of My Files too.
But indeed this feature would be a truly useful addition and a big timesaver.
Especially with recursive folder up/downloading or server side zip/unzip. (separate issues here I belive)

**Comment by yomono** [comment]
Let me revive it with you then... hehe

Honestly, the feature is 95% there, it would be nice to implement it

**Comment by folosleg** [comment]
I don't know the codebase at all, so I personally wouldn't go into percentages...maybe you do...
But sometimes what seems to be that 5% is really the difference between the human mind and computer logic.
Maybe a couple lines of code, maybe a few thousand.
Much respect for the devs, they do a great job with MC. Hopefully they can implement this as well.
If @Ylianst needs any user input / exact usecase on this one, happy to participate here.

**Comment by yomono** [comment]
> I don't know the codebase at all, so I personally wouldn't go into percentages...maybe you do... But sometimes what seems to be that 5% is really the difference between the human mind and computer logic. Maybe a couple lines of code, maybe a few thousand. Much respect for the devs, they do a great job with MC. Hopefully they can implement this as well. If @Ylianst needs any user input / exact usecase on this one, happy to participate here.

The reason I say 95% is not because I "feel it", but because there is a very similar implementation with the plugin called "Files distribution" (check the link in the first post). That plugin basically already does this, but instead of doing it manually, it will trigger a push every X minutes. So the functionality it's already there, just implemented in a different way.
And yes, Ylianst and team are amazing!!

**Comment by openncomp** [comment]
Yes, it would be a great feature to send files from "Server files" to remote devices.
